fml/paths.h

```diff
@@ -13,6 +13,7 @@
 namespace fml {
 namespace paths {
 
+std::pair<bool, std::string> GetExecutablePath();
```

Comment:
I think (in a later patch maybe) we should refactor these to use std::optional now that it is available to us.

Reply:
Can we dry this up a little bit by putting `GetExecutableDirectoryPath` in the common implementation (It'll just call `GetDirectoryName(GetExecutablePath())` after an optional value check) and having platform specific implementations of `GetExecutablePath()`?

Reply:
done